simple_parsing/replace.py

```diff
@@ -90,3 +104,80 @@ def replace(obj: DataclassT, changes_dict: dict[str, Any] | None = None, **chang
     replace_kwargs.update(changes)
 
     return dataclasses.replace(obj, **replace_kwargs)
+
+
+def unflatten_selection_dict(
+    flattened: Mapping[str, V], keyword: str = "__key__", sep="."
+) -> PossiblyNestedDict[str, V]:
+    """
+    This function convert a flattened dict into a nested dict
+    and it inserts the `keyword` as the selection into the nested dict.
+
+    >>> unflatten_selection_dict({'ab_or_cd': 'cd', 'ab_or_cd.c_or_d': 'd'})
+    {'ab_or_cd': {'__key__': 'cd', 'c_or_d': 'd'}}
+
+    >>> unflatten_selection_dict({"a": 1, "b": 2})
+    {'a': {'__key__': 1}, 'b': {'__key__': 2}}
```

Comment:
I don't understand this case here. There isn't any overlap between the keys, so shouldn't it return the same dict unchanged?

Reply:
Fixed.

Reply:
Fixed the unflatten_selection_dict and add recursive flag to it. It passed those tests:
```python
    >>> unflatten_selection_dict({'ab_or_cd': 'cd', 'ab_or_cd.c_or_d': 'd'})
    {'ab_or_cd': {'__key__': 'cd', 'c_or_d': 'd'}}
    
    >>> unflatten_selection_dict({'lv1': 'a', 'lv1.lv2': 'b', 'lv1.lv2.lv3': 'c'})
    {'lv1': {'__key__': 'a', 'lv2': 'b', 'lv2.lv3': 'c'}}

    >>> unflatten_selection_dict({'lv1': 'a', 'lv1.lv2': 'b', 'lv1.lv2.lv3': 'c'}, recursive=True)
    {'lv1': {'__key__': 'a', 'lv2': {'__key__': 'b', 'lv3': 'c'}}}

    >>> unflatten_selection_dict({'ab_or_cd.c_or_d': 'd'})
    {'ab_or_cd': {'c_or_d': 'd'}}

    >>> unflatten_selection_dict({"a": 1, "b": 2})
    {'a': 1, 'b': 2}
```